### DISCUSSION TITLE
Github.action_path contains `\` on windows runners even when the shell is set to bash

### DISCUSSION THREAD

**Discussion by SaschaMann** [community/community#25910] [Actions]
<p>I’m trying to use <code>${{ github.action_path }}</code> in a <a href="https://github.com/julia-actions/julia-processcoverage/blob/736914d23121833fe52b6b047c4c9777ddb693cf/action.yml" rel="nofollow noopener">composite action</a>.</p>
<p>When I run a workflow containing this action on a Windows runner, the path contains backslashes, even if the shell is set to <code>bash</code>:</p>
<pre><code class="lang-auto">##[debug]Evaluating: format('julia --color=yes {0}/main.jl', github.action_path)
##[debug]Evaluating format:
##[debug]..Evaluating String:
##[debug]..=&gt; 'julia --color=yes {0}/main.jl'
##[debug]..Evaluating Index:
##[debug]....Evaluating github:
##[debug]....=&gt; Object
##[debug]....Evaluating String:
##[debug]....=&gt; 'action_path'
##[debug]..=&gt; 'd:\a\_actions\julia-actions\julia-processcoverage\composite'
##[debug]=&gt; 'julia --color=yes d:\a\_actions\julia-actions\julia-processcoverage\composite/main.jl'
##[debug]Result: 'julia --color=yes d:\a\_actions\julia-actions\julia-processcoverage\composite/main.jl'
##[debug]Loading env
##[debug]Run julia --color=yes d:\a\_actions\julia-actions\julia-processcoverage\composite/main.jl
##[debug]julia --color=yes d:\a\_actions\julia-actions\julia-processcoverage\composite/main.jl
##[debug]shell: C:\Program Files\Git\bin\bash.EXE --noprofile --norc -e -o pipefail {0}
##[debug]
##[debug]C:\Program Files\Git\bin\bash.EXE --noprofile --norc -e -o pipefail d:\a\_temp\574911a6-bba0-4689-9ca9-7ed9a1eaba1c.sh
ERROR: could not open file d:\a\StringBuilders.jl\StringBuilders.jl\a_actionsjulia-actionsjulia-processcoveragecomposite\main.jl
</code></pre>
<p><a href="https://github.com/SaschaMann/StringBuilders.jl/pull/1/checks?check_run_id=959070418" rel="nofollow noopener">Example build</a></p>
<p>This makes the step fail, because the backslashes aren’t escaped properly to be used in bash.</p>
<p>What is the recommended way to use <code>github.action_path</code> in a cross-platform way while using bash as shell for the step?</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/saschamann">@saschamann</a>,</p>
<p>Please use <code>$GITHUB_ACTION_PATH</code>  instead.</p>
<pre><code class="lang-auto">    - run: julia --color=yes $GITHUB_ACTION_PATH/main.jl
      shell: bash
</code></pre>
<p>Doc here: <a href="https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#runsstepsrun" rel="nofollow noopener">https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#runsstepsrun</a></p>
<p>Thanks.</p>

**Comment by SaschaMann** [comment]
<p>That worked, thank you :slight_smile:</p>
